### ISSUE TITLE
Non-deterministic seed finding

### ISSUE THREAD

**Issue by beomki-yeo** [acts-project/traccc#163]
Some people might have already noticed that the seeding result is not deterministic for every rum time.
The non-determinism comes from the fact that randomly sorted triplets from `triplet_finding` will get different weights in `weight_updating` kernel function.  Seed filtering, which includes `weight_updating`, is sensitive to the order of triplets. 

If people hate non-determinism, there are two ways to resolve it.

1. Sort triplets after triplet_finding but it will cost a lot

2. Make the `weight_updating` output independent of the order of triplets. 

Any help on this issue will be appreciated. Also please feel free to post your opinion on this.

**Comment by stephenswat** [comment]
My personal opinion is that having non-deterministic outcomes here is okay. In essence, the seed finding algorithm we used on the CPU was already non-deterministic (or at least sensitive to the order of execution) and this become noticeable when we took the code to a non-deterministically scheduled environment. Therefore, I don't personally think there is any problem with your implementation, but rather with the algorithm itself.

Maybe we can add an optional sorting step which we would turn on for testing purposes, but which we would then turn off for any real-world processing?
